src/libcmd/common-eval-args.cc

```diff
@@ -197,12 +197,13 @@ SourcePath lookupFileArg(EvalState & state, std::string_view s, const Path * bas
     }
 
     else if (s.size() > 2 && s.at(0) == '<' && s.at(s.size() - 1) == '>') {
-        Path p(s.substr(1, s.size() - 2));
+        // Should perhaps be a `CanonPath`?
+        std::string p(s.substr(1, s.size() - 2));
         return state.findFile(p);
     }
 
     else
-        return state.rootPath(baseDir ? absPath(s, *baseDir) : absPath(s));
+        return state.rootPath(absPath(baseDir ? *baseDir / s : s).string());
```

Comment:
```suggestion
        return state.rootPath((baseDir ? absPath(s, *baseDir) : absPath(s)).string());
```
with the changes we made before, can do this which is closer to the original.

Reply:
oh yeah great point, we no longer have the pointer vs optional dance to worry about!